lib/landrush/landrush_ip.rb

```diff
@@ -0,0 +1,176 @@
+
+          platform = os[0].to_s.to_sym
+        end
+      end
+
+      if platform.nil? or !@platforms.has_key?(platform)
+        @error = "Unsupported guest platform: #{platform}"
+
+        return false
+      end
+
+      # Now let's check the architecture
+      if arch.nil?
+        case platform
+        when :windows
+          #
+          # See:
+          # - http://superuser.com/a/293143
+          # - http://blogs.msdn.com/b/david.wang/archive/2006/03/26/howto-detect-process-bitness.aspx
+          #
+          # Quicker than using systeminfo and works from XP and up.
+          #
+          script = <<-EOH
+            set Arch=x64
+            if "%PROCESSOR_ARCHITECTURE%" == "x86" (
+                if not defined PROCESSOR_ARCHITEW6432 set Arch=x86
+            )
+            echo %Arch%
+          EOH
+
+          result = ''
+          @machine.communicate.execute(script) do |type, data|
+            result << data if type == :stdout
+          end
+
+          if result =~ /(x64)/i
+            arch = 'amd64'
+          elsif result =~ /(x86)/i
+            arch = '386'
+          end
+        else
+          #
+          # Windows aside, every supported UNIX flavour includes uname
+          # For once there's a UNIX tool that actually behaves relatively consistently across platforms.
+          #
+          result = ''
+          @machine.communicate.execute('uname -mrs') do |type, data|
+            result << data if type == :stdout
+          end
+
+          #
+          # uname -mrs will return the following architectures:
+          # i386 i686 x86_64 ia64 alpha amd64 arm armeb armel hppa m32r m68k
+          # mips mipsel powerpc ppc64 s390 s390x sh3 sh3eb sh4 sh4eb sparc
+          #
+          # The vast majority are irrelevant to us.
+          #
+          if result =~ /(x86_64|ia64|amd64)/i
+            arch = 'amd64'
+          elsif result =~ /(arm)/i
+            arch = 'arm'
+          elsif result =~ /(i386|i686)/i
+            arch = '386'
+          end
+        end
+      end
+
+      if arch.nil? or !@platforms[platform].include?(arch)
+        @error = "Unsupported guest architecture: #{arch} (#{platform})"
+
+        return false
+      end
+
+      # We've got platform and architecture now
+      info "Platform: #{platform}/#{arch}"
+
+      @machine.communicate.tap do |comm|
+        ssh_info = nil
+        retryable(on: Vagrant::Errors::SSHNotReady, tries: 3, sleep: 2) do
+          ssh_info = @machine.ssh_info
+          raise Vagrant::Errors::SSHNotReady if ssh_info.nil?
+        end
+
+        host_path = @machine.env.tmp_path.join("#{@machine.id}-landrush-ip")
+        host_path.delete if host_path.file?
+
+        begin
+          http             = Net::HTTP.new('api.github.com', 443)
+          http.use_ssl     = true
+          http.verify_mode = OpenSSL::SSL::VERIFY_PEER
+
+          response = http.request(Net::HTTP::Get.new('/repos/werelds/landrush-ip/releases/latest'))
```

Comment:
do we really want to include latest? Would it not be better to use an explicit tag? Otherwise behavior of a given Landrush version might change, if landrush-ip gets a new release which can change to hard to track problems?  WDYT?